cloud/amazon/route53_health_check.py

```diff
@@ -0,0 +1,351 @@
+        params['string_match_part'] = HealthCheck.XMLStringMatchPart % {'string_match': health_check.string_match}
+    return params
+
+XMLResourcePathPart = """<ResourcePath>%(resource_path)s</ResourcePath>"""
+
+POSTXMLBody = """
+    <CreateHealthCheckRequest xmlns="%(xmlns)s">
+        <CallerReference>%(caller_ref)s</CallerReference>
+        <HealthCheckConfig>
+            %(ip_addr_part)s
+            <Port>%(port)s</Port>
+            <Type>%(type)s</Type>
+            %(resource_path_part)s
+            %(fqdn_part)s
+            %(string_match_part)s
+            <RequestInterval>%(request_interval)s</RequestInterval>
+            <FailureThreshold>%(failure_threshold)s</FailureThreshold>
+        </HealthCheckConfig>
+    </CreateHealthCheckRequest>
+    """
+
+UPDATEHCXMLBody = """
+    <UpdateHealthCheckRequest xmlns="%(xmlns)s">
+        <HealthCheckVersion>%(health_check_version)s</HealthCheckVersion>
+        %(ip_addr_part)s
+        <Port>%(port)s</Port>
+        %(resource_path_part)s
+        %(fqdn_part)s
+        %(string_match_part)s
+        <FailureThreshold>%(failure_threshold)i</FailureThreshold>
+    </UpdateHealthCheckRequest>
+    """
+
+def create_health_check(conn, health_check, caller_ref = None):
+    if caller_ref is None:
+        caller_ref = str(uuid.uuid4())
+    uri = '/%s/healthcheck' % conn.Version
+    params = to_template_params(health_check)
+    params.update(xmlns=conn.XMLNameSpace, caller_ref=caller_ref)
+
+    xml_body = POSTXMLBody % params
+    response = conn.make_request('POST', uri, {'Content-Type': 'text/xml'}, xml_body)
+    body = response.read()
+    boto.log.debug(body)
+    if response.status == 201:
+        e = boto.jsonresponse.Element()
+        h = boto.jsonresponse.XmlHandler(e, None)
+        h.parse(body)
+        return e
+    else:
+        raise exception.DNSServerError(response.status, response.reason, body)
+
+def update_health_check(conn, health_check_id, health_check_version, health_check):
+    uri = '/%s/healthcheck/%s' % (conn.Version, health_check_id)
+    params = to_template_params(health_check)
+    params.update(
+        xmlns=conn.XMLNameSpace,
+        health_check_version=health_check_version,
+    )
+    xml_body = UPDATEHCXMLBody % params
+    response = conn.make_request('POST', uri, {'Content-Type': 'text/xml'}, xml_body)
+    body = response.read()
+    boto.log.debug(body)
+    if response.status not in (200, 204):
+        raise exception.DNSServerError(response.status,
+                                       response.reason,
+                                       body)
+    e = boto.jsonresponse.Element()
+    h = boto.jsonresponse.XmlHandler(e, None)
+    h.parse(body)
+    return e
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+            state               = dict(required=False, choices=['present', 'absent'], default='present'),
+            ip_address          = dict(required=False),
```

Comment:
this (and other similar lines) can just be `dict()` - `required=False` is implied.